modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/ViewChangesDisplayContext.java

```diff
@@ -1612,11 +1614,36 @@ private <T extends BaseModel<T>> void _populateEntryValues(
 						(ctSQLMode == CTSQLModeThreadLocal.CTSQLMode.DEFAULT)) {
 
 						if (ctModelMap == null) {
-							ctModelMap =
-								_ctDisplayRendererRegistry.fetchCTModelMap(
-									_ctCollection.getCtCollectionId(),
-									CTSQLModeThreadLocal.CTSQLMode.DEFAULT,
-									modelClassNameId, classPKs);
+							int i = 0;
+
+							while (i < classPKs.size()) {
+								int endIndex = i + _SQL_SERVER_PARAM_LIMIT;
+
+								if (endIndex > classPKs.size()) {
+									endIndex = classPKs.size() - 1;
+								}
+
+								Long[] batchClassPKs = ArrayUtil.subset(
+									classPKs.toArray(new Long[0]), i, endIndex);
+
+								Map<Serializable, T> batchCtModelMap =
+									_ctDisplayRendererRegistry.fetchCTModelMap(
+										_ctCollection.getCtCollectionId(),
+										CTSQLModeThreadLocal.CTSQLMode.DEFAULT,
+										modelClassNameId,
+										SetUtil.fromArray(batchClassPKs));
+
+								if (batchCtModelMap != null) {
+									if (ctModelMap == null) {
+										ctModelMap = batchCtModelMap;
+									}
+									else {
+										ctModelMap.putAll(batchCtModelMap);
+									}
+								}
+
+								i += _SQL_SERVER_PARAM_LIMIT;
+							}
```

Comment:
@noornajj shouldn't we make this change in fetchCTModelMap?

Reply:
actually it looks like the code is just calling fetchByPrimaryKeys.  Why isn't that one properly splitting the query?

Reply:
I looked into this and found that we need to set a portal property for this to work. I think we can discard the ticket in this case

Reply:
I actually ran into the same error though after resolving this one in CTConflictChecker. We already have the query batched into separate sets of "in", but this will still fail for SQL Server because they take into account the total number of parameters in the query. I'll rewrite the bug to reflect this and send a fix for it.